From 43d024168a4fa2aca3de1089a2a836f4136aac26 Mon Sep 17 00:00:00 2001
From: Mahesh Kumar P <mahesh.kumar.p@intel.com>
Date: Mon, 20 Jul 2015 16:01:00 +0530
Subject: [PATCH] pwm: Fix the PWM suspend flow on CHT platform  according to
 BIOS changes

The patch fixes a bug in the suspend flow in CHT platform.The
ACPI power state was compared to a wrong value because of a BUG in BIOS
code.

The PWM entry in BIOS is getting fixed and this
patch needs to get merged with the BIOS changes to get PWM fully fixed
on CHT platforms.

Change-Id: Ia9b7f39d8e5dcffbb69f35ca9700b2a21bf5a3ac
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-13790
Signed-off-by: Kumar P, Mahesh <mahesh.kumar.p@intel.com>
---
 drivers/misc/pwm/pwm_byt_core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/misc/pwm/pwm_byt_core.c b/drivers/misc/pwm/pwm_byt_core.c
index 509e504f3d94..490aa621cdb7 100644
--- a/drivers/misc/pwm/pwm_byt_core.c
+++ b/drivers/misc/pwm/pwm_byt_core.c
@@ -552,7 +552,7 @@ static int pwm_byt_suspend(struct device *dev)
 		return r;
 	}
 
-	if (!adev->power.state) {
+	if (adev->power.state) {
 		pr_info("Device %s is already in non D0 state\n",
 				dev_name(dev));
 		return r;
-- 
1.9.1

